File: python/입문/가까운_수.py
# 가까운 수 문제

def solution(array, n):
    array.sort()
    answer = array[0]
    diff = abs(answer - n)
    for i in array:
        d = abs(i - n)
        if diff > d:
            diff = d
            answer = i
    return answer

File: java/입문/가까운_수.java
class Solution {
    public int solution(int[] array, int n) {
        int answer = array[0];
        int min = Math.abs(n - array[0]);
        for(int i = 1; i < array.length; i++) {
            int diff = Math.abs(n - array[i]);
            if (min > diff || (min == diff && answer > array[i])) {
                answer = array[i];
                min = diff;
            }
        }
        return answer;
    }
}